src/SDK/Common/Util/DanglingSpansShutdownHandler.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Util;
+
+use \OpenTelemetry\Context\DebugScope;
+use OpenTelemetry\SDK\Trace\Context;
+
+/**
+ * SDK helper classes for dealing with dangling spans
+ */
+final class DanglingSpansShutdownHandler
+{
+    /**
+     * Function to remove any contexts that may have been left over by early exits where
+     * scope->detach was not called. e.g. An early exit() call from an PHP script/request.
+     *
+     * To register this call:
+     * ShutdownHandler::register(DanglingSpansShutdownHandler::shutdown)
+     *
+     * BUT make sure you do this before you register the other shutdown handlers or any spans won't be processed/exported
+     *
+     * @return void
+     */
+    public static function shutdown(callable $fn): void
+    {
+        $ctx = Context::storage();
+        while ($ctx->current() !== Context::getRoot()) {
+            $span = Span::fromContext($ctx->current());
+            $span->end();
+            $ds = new DebugScope($ctx->scope());
```

Comment:
what's the benefit of using debugscope compared to just detaching the active scope?

Reply:
If I don't wrap it in a debugScope it won't set the correct properties and will trigger_error when the original DebugScope goes out of... scope.

Reply:
It doesn't seem right. We don't need to do that ordinarily (eg how all/most auto-instrumentation post hooks end the active span).

Reply:
I thought about this a bit more, and I see the issue. DebugScope keeps a reference to a scope, but doesn't store it anywhere. That means currently the expectation is that you'd keep a reference to the debugscope, and deactivate *that*, rather than the "real" scope.
I think that's actually also the issue in #1138 

TBH I'm surprised that just creating a new debug scope and then deactivating it actually works, it's surely by happy accident. I'm now wondering whether DebugScope can be improved somewhat, since in this case it's complaining that a scope hasn't been detached when in fact it has. I'll have a play with this - perhaps a weak reference to the real scope, or the debugscope itself could be stored?